### PR TITLE
Feature/eip1 4288 open api specification for postal/proxy approved application

### PR DESCRIPTION
**Pull request by spalamthodi**
Open API Specification for Approved postal/proxy vote application
https://technologyprogramme.atlassian.net/browse/EIP1-4287
https://technologyprogramme.atlassian.net/browse/EIP1-4288

### CODE REVIEW

File: src/main/resources/openapi/sqs/approved-postal-vote-application-sqs-messaging.yaml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+              street: 234 Kingsway
+              locality: Camden
+              town: London
+              area: Greater London
+              postcode: SW112DR
+              uprn: '098766'
+            ballotAddressReason: I am away from home for work
+            signature: LzlqLzRBQ...
+            postalVoteUntilFurtherNotice: true
+          signature: LzlqLzRBQ...
```
Comment (inderjitvirdi) [inline src/main/resources/openapi/sqs/approved-postal-vote-application-sqs-messaging.yaml:120]: Duplicate signature field - prob the one in postalVoteDetails can be removed?

File: src/main/resources/openapi/sqs/approved-proxy-vote-application-sqs-messaging.yaml [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+              area: Greater London
+              postcode: SW112DR
+              uprn: 098765
+            phone: 07799779977
+            email: john@example.com
+            refNum: V3HQ9LKI0W
+            ip: 1.2.3.4
+            lang: en
+            emsElectorId: e87cbaea-0deb-4058-95c6-8240d426f5e1
+          proxyDetails:
```
Comment (inderjitvirdi) [inline src/main/resources/openapi/sqs/approved-proxy-vote-application-sqs-messaging.yaml:138]: call this proxyVoteDetails?

**Review by kristerbone — APPROVED** [review]
LGTM, nice one.

File: src/main/resources/openapi/sqs/approved-proxy-vote-application-sqs-messaging.yaml (hunk 2 above)
Comment (spalamthodi) [reply src/main/resources/openapi/sqs/approved-proxy-vote-application-sqs-messaging.yaml]: Done

File: src/main/resources/openapi/sqs/approved-postal-vote-application-sqs-messaging.yaml (hunk 1 above)
Comment (spalamthodi) [reply src/main/resources/openapi/sqs/approved-postal-vote-application-sqs-messaging.yaml:119]: Removed

File: src/main/resources/openapi/sqs/approved-proxy-vote-application-sqs-messaging.yaml [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        $ref: '#/components/requestBodies/ApprovedProxyVoteApplicationMessage'
+      responses:
+        '204':
+          description: No response content.
+components:
+  #
+  # Schema and Enum Definitions
+  # -
+  schemas:
+    ProxyDetails:
```
[the remark this answers is not in the document]
Comment (spalamthodi) [reply src/main/resources/openapi/sqs/approved-proxy-vote-application-sqs-messaging.yaml]: Done

File: src/main/resources/openapi/sqs/approved-proxy-vote-application-sqs-messaging.yaml [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+              street: 125 Kingsway
+              locality: Camden
+              town: London
+              area: Greater London
+              postcode: SW112DR
+              uprn: 098765
+            proxyPhone: '07777123456'
+            proxyEmail: jane@example.com
+            proxyReason: work commitments
+            signature: LzlqLzRBQ...
```
[the remark this answers is not in the document]
Comment (spalamthodi) [reply src/main/resources/openapi/sqs/approved-proxy-vote-application-sqs-messaging.yaml]: Done

File: src/main/resources/openapi/sqs/approved-proxy-vote-application-sqs-messaging.yaml [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+            proxyVoteUntilFurtherNotice: true
+
+      properties:
+        approvalDetails:
+          $ref: '../sqs/Models/ApprovalDetails.yaml'
+        applicantDetails:
+          $ref: '../sqs/Models/ApplicantDetails.yaml'
+        proxyDetails:
```
[the remark this answers is not in the document]
Comment (spalamthodi) [reply src/main/resources/openapi/sqs/approved-proxy-vote-application-sqs-messaging.yaml]: Done

File: src/main/resources/openapi/sqs/approved-proxy-vote-application-sqs-messaging.yaml [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+          $ref: '#/components/schemas/ProxyDetails'
+        signature:
+          type: string
+          description: ' an image of the applicant’s signature, in Base64 encoded PNG format.The signature image will be a Base64 string encoding a PNG image. The image will be 1050 pixels wide and 300 pixels high, corresponding to a scan of the paper forms at 300 dpi. The image will be black and white only. The image will be cropped so that the signature fills the box with a small white margin. The data length is usuall'
+      required:
+        - approvalDetails
+        - applicantDetails
+        - proxyDetails
```
[the remark this answers is not in the document]
Comment (spalamthodi) [reply src/main/resources/openapi/sqs/approved-proxy-vote-application-sqs-messaging.yaml]: Done

File: src/main/resources/openapi/sqs/Models/Address.yaml [hunk 7]
```diff
@@ -0,0 +1,41 @@
+title: Address
+description: Address format used throughout gov.uk voter services.
+type: object
```
[the remark this answers is not in the document]
Comment (spalamthodi) [reply src/main/resources/openapi/sqs/Models/Address.yaml:3]: Done

**Review by inderjitvirdi — APPROVED** [review]

**Pull request merged**